linter.py

```diff
@@ -20,41 +7,23 @@
 
 
 class JSON(Linter):
-    """Provides an interface to json.loads()."""
-
-    syntax = 'json'
     cmd = None
     loose_regex = re.compile(r'^.+: (?P<message>.+) in \(data\):(?P<line>\d+):(?P<col>\d+)')
     strict_regex = re.compile(r'^(?P<message>.+):\s*line (?P<line>\d+) column (?P<col>\d+)')
     regex = loose_regex
     defaults = {
+        'selector': 'source.json',
         'strict': True
     }
 
-    extensions = [
-        '.sublime-build',
-        '.sublime-color-scheme',
-        '.sublime-commands',
-        '.sublime-completions',
-        '.sublime-keymap',
-        '.sublime-menu',
-        '.sublime_metrics',
-        '.sublime-mousemap',
-        '.sublime-project',
-        '.sublime_session',
-        '.sublime-settings',
-        '.sublime-theme',
-        '.sublime-workspace',
-    ]
-
     def run(self, cmd, code):
         """
         Attempt to parse code as JSON.
 
         Returns '' if it succeeds, the error message if it fails.
         Use ST's loose parser for its setting files, or when specified.
         """
-        is_sublime_file = os.path.splitext(self.filename)[1] in self.extensions
+        is_sublime_file = os.path.splitext(self.filename)[1].startswith('.sublime-')
 
         if Linter.get_view_settings(self).get('strict') and not is_sublime_file:
```

Comment:
Linter.get_view_settings(self) -> self.get_view_setting()

Also below: mutating `self.__class__.regex` is not allowed. Should try `self.regex`.

Reply:
Better now?